File: Air Strike/Air Strike/FoePlane.m
//
//  FoePlane.m
//  Air Strike
//
//  Created by Chang Tong Xue on 2015-12-17.
//  Copyright © 2015 DX. All rights reserved.
//

#import "FoePlane.h"

@implementation FoePlane

+ (instancetype)createBigPlane{
    FoePlane *foePlane = (FoePlane *)[FoePlane spriteNodeWithTexture:[SharedTexture textureWithType:TextureTypeBigFoePlane]];
    foePlane.hp = 7;
    foePlane.type = FoePlaneTypeBig;
    foePlane.physicsBody = [SKPhysicsBody bodyWithRectangleOfSize:foePlane.size];
    return foePlane;
}

+ (instancetype)createMediumPlane{
    FoePlane *foePlane = (FoePlane *)[FoePlane spriteNodeWithTexture:[SharedTexture textureWithType:TextureTypeMediumFoePlane]];
    foePlane.hp = 5;
    foePlane.type = FoePlaneTypeMedium;
    foePlane.physicsBody = [SKPhysicsBody bodyWithRectangleOfSize:foePlane.size];
    return foePlane;
}

+ (instancetype)createSmallPlane{
    FoePlane *foePlane = (FoePlane *)[FoePlane spriteNodeWithTexture:[SharedTexture textureWithType:TextureTypeSmallFoePlane]];
    foePlane.hp = 1;
    foePlane.type = FoePlaneTypeSmall;
    foePlane.physicsBody = [SKPhysicsBody bodyWithRectangleOfSize:foePlane.size];
    return foePlane;
}


@end

File: Air Strike/Air Strike/FoePlane.swift
//
//  FoePlane.swift
//  Air Strike
//
//  Created by Chang Tong Xue on 2016-01-01.
//  Copyright © 2016 DX. All rights reserved.
//

import SpriteKit

enum FoePlaneType: Int {
    case Big = 1
    case Medium
    case Small
}

class FoePlane: SKSpriteNode {

    var hp: Int = 1
    var type: FoePlaneType = .Small
    
    class func createSmallPlane() -> FoePlane {
        let plane = FoePlane(texture: SharedAtlas.textureWithType(.SmallFoePlane))
        plane.physicsBody = SKPhysicsBody(rectangleOfSize: plane.size)
        return plane
    }
    
    class func createMediumPlane() -> FoePlane {
        let plane = FoePlane(texture: SharedAtlas.textureWithType(.MediumFoePlane))
        plane.hp = 5
        plane.type = .Medium
        plane.physicsBody = SKPhysicsBody(rectangleOfSize: plane.size)
        return plane
    }
    
    class func createBigPlane() -> FoePlane {
        let plane = FoePlane(texture: SharedAtlas.textureWithType(.BigFoePlane))
        plane.hp = 7
        plane.type = .Big
        plane.physicsBody = SKPhysicsBody(rectangleOfSize: plane.size)
        return plane
    }

}



